### PR TITLE
Masking and clipping articles

### LINKED ISSUE #3669: content: migrate CSS Masking from HTML5 Rocks
#### Provide a brief title for your content proposal

CSS Masking

#### Provide a brief summary of your content proposal

Rewrite this outdated guide with some new examples and links out to other good material for people to explore in depth.

https://html5rocks.com/en/tutorials/masking/adobe/

#### Who will be authoring the content?

@rachelandrew 

#### What area is your content related to?

Choose from one of the following: Frontend

#### What kind of content are you creating?

Choose from one of the following: Blog Post

### PR DESCRIPTION
**Pull request by rachelandrew**
Fixes #3669 and #3691 

Changes proposed in this pull request:

I have written two new articles to replace https://html5rocks.com/en/tutorials/masking/adobe/ 

- Adds CSS Clipping post
- Adds CSS Masking post
- Fixes a couple of duplicate IDs in other posts which were stopping eleventy building.

### CODE REVIEW

File: src/site/content/en/blog/css-masking/index.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+You will need to use the `-webkit-` prefix in addition to the standard property in order to achieve the best browser compatibility.
+See [Can I use CSS Masks?](https://caniuse.com/#feat=css-masks) for full browser support information.
+
+While browser support using the prefixed property is good,
+when using masking to make text on top of an image visible take care of what will happen if masking is unavailable.
+It may be worth using feature queries to detect support for `mask-image` or `-webkit-mask-image`
+and providing a readable fallback before adding your masked version.
+
+```css
+@supports(-webkit-mask-image: url(#mask)) or (mask-image: url(#mask)); {
```
Comment (kaycebasques) [inline src/site/content/en/blog/css-masking/index.md:35]:
```suggestion
@supports(-webkit-mask-image: url(#mask)) or (mask-image: url(#mask)) {
```
I'm not familiar with `@supports` but that semicolon seems out-of-place?

File: src/site/content/en/blog/css-masking/index.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+The `mask-image` property works in a similar way to the `background-image` property.
+Use a `url()` value to pass in an image.
+Your mask image needs to have a transparent or semi-transparent area.
+
+A fully transparent area will cause the part of the image under that area to be invisible.
+Using an area which is semi-transparent however will allow some of the original image to show through.
+You can see the difference in the Glitch below.
+The first image is the original image of balloons with no mask.
+The second image has a mask applied which has a white star on a fully transparent background.
+The third image has a white star on a background with a gradient transparency.
```
Comment (kaycebasques) [inline src/site/content/en/blog/css-masking/index.md:51]: Very clear, thanks

File: src/site/content/en/blog/css-masking/index.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
+---
+title: How to use CSS masking
+description: |
+  CSS masking gives you the option of using an image as a mask layer.
+  This means that use can use an image, an SVG, or a gradient as your mask,
+  to create interested effects without an image editor.
+authors:
+  - rachelandrew
+date: 2020-09-02
+hero: hero.jpg
```
Comment (kaycebasques) [inline src/site/content/en/blog/css-masking/index.md:10]: Can you poke around on unsplash.com and find some other kind of metaphor to use here? I want to avoid reminding people of covid19.

File: src/site/content/en/blog/sign-in-form-best-practices/index.md [hunk 4]
```diff
@@ -523,7 +523,7 @@ to use `#ccc` or darker.
 </figure>
 
 
-### Use built-in browser features to warn of invalid input values {: #validation }
+### Use built-in browser features to warn of invalid input values {: #browserfeatures }
```
Comment (kaycebasques) [inline src/site/content/en/blog/sign-in-form-best-practices/index.md:526]:
```suggestion
### Use built-in browser features to warn of invalid input values {: #validation }
```

File: src/site/content/en/blog/sign-in-form-best-practices/index.md [hunk 5]
```diff
@@ -50,9 +50,9 @@ Here is an example of a simple sign-in form that demonstrates all of the best pr
 * Use element attributes to [access built-in browser
```
Comment (kaycebasques) [inline src/site/content/en/blog/sign-in-form-best-practices/index.md:50]: Another unrelated file was modified. This one is mostly OK because it's just removing trailing whitespace, but there is one thing we need to fix.

File: src/site/content/en/blog/hid-examples/index.md [hunk 6]
```diff
@@ -136,7 +136,7 @@ how to use the API yourself.
 * [TAG Review](https://github.com/w3ctag/design-reviews/issues/370)
 * [Intent to Prototype](https://groups.google.com/a/chromium.org/g/blink-dev/c/LoyzK8xTRME/m/yLR-K-siBQAJ)
 
-## Acknowledgements {: #helpful }
+## Acknowledgements {: #ack }
```
Comment (kaycebasques) [inline src/site/content/en/blog/hid-examples/index.md:139]:
```suggestion
## Acknowledgements {: #helpful   }
```

Unrelated file was modified

File: src/site/content/en/blog/css-clipping/index.md [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+<div class="glitch-embed-wrap" style="height: 420px; width: 100%;">
+  <iframe
+    src="https://glitch.com/embed/#!/embed/clip-path?path=README.md&previewSize=100"
+    title="clip-path on Glitch"
+    allow="encrypted-media"
+    style="height: 100%; width: 100%; border: 0;">
+  </iframe>
+</div>
+
+In the example above, the balloon image is square.
```
Comment (kaycebasques) [inline src/site/content/en/blog/css-clipping/index.md:31]:
```suggestion
In the example above, the balloon image is square ([source](https://cdn.glitch.com/5984697d-c5e8-4a63-a765-7e7ac3916fc0%2Fround-balloon.jpg?v=1597216988521)).
```

File: src/site/content/en/blog/css-clipping/index.md [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+
+Clipping can be applied to any HTML element, not just images.
+There are a few different ways to create a `clip-path`, in this post we will take a look at them.
+
+## Browser compatibility
+
+Other than the box values as explained later in the post,
+the various values of `clip-path` demonstrated have [excellent browser support](https://caniuse.com/#feat=css-clip-path).
+For legacy browsers a fallback may be to allow the browser to ignore the `clip-path` property and show the unclipped image.
+If this is a problem you could test for `clip-path` in a feature query and offer an alternate layout for unsupporting browsers.
```
Comment (kaycebasques) [inline src/site/content/en/blog/css-clipping/index.md:47]: Just realizing something… is `@supports` itself supported in legacy browsers? If it's IE11 then I suppose not… https://caniuse.com/mdn-css_at-rules_supports_selector

File: src/site/content/en/blog/css-clipping/index.md [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+@supports(clip-path: circle(45%) {
+  /* code that requires clip-path here. */
+}
+```
+
+## Basic shapes
+
+The `clip-path` property can take a number of values.
+The value used in the initial example was `circle()`.
+This is one of the basic shape values, which are defined in the CSS Shapes specification.
```
Comment (kaycebasques) [inline src/site/content/en/blog/css-clipping/index.md:59]: Link to spec please

File: src/site/content/en/blog/css-clipping/index.md [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+Note that CSS Shapes can only be applied to floated elements.
+The `clip-path` property does not require the element to be floated.
+{% endAside %}
+
+The full list of basic shapes is:
+
+- [inset()](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Shapes/Basic_Shapes#inset)
+- [circle()](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Shapes/Basic_Shapes#circle)
+- [ellipse()](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Shapes/Basic_Shapes#ellipse)
+- [polygon()](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Shapes/Basic_Shapes#polygon)
```
Comment (kaycebasques) [inline src/site/content/en/blog/css-clipping/index.md:82]:
```suggestion
- [`inset()`](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Shapes/Basic_Shapes#inset)
- [`circle()`](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Shapes/Basic_Shapes#circle)
- [`ellipse()`](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Shapes/Basic_Shapes#ellipse)
- [`polygon()`](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Shapes/Basic_Shapes#polygon)
```

File: src/site/content/en/blog/css-clipping/index.md [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+
+### The inset() value
```
Comment (kaycebasques) [inline src/site/content/en/blog/css-clipping/index.md:84]:
```suggestion
### `inset()`
```
Making it consistent with the sections below (which only say the method name)

File: src/site/content/en/blog/css-clipping/index.md [hunk 12]
```diff
@@ -0,0 +1,223 @@
...
+<div class="glitch-embed-wrap" style="height: 420px; width: 100%;">
+  <iframe
+    src="https://glitch.com/embed/#!/embed/clip-path-inset?path=README.md&previewSize=100"
+    title="clip-path-inset on Glitch"
+    allow="encrypted-media"
+    style="height: 100%; width: 100%; border: 0;">
+  </iframe>
+</div>
+
+### The circle() value
```
Comment (kaycebasques) [inline src/site/content/en/blog/css-clipping/index.md:103]:
```suggestion
### `circle()`
```

File: src/site/content/en/blog/css-clipping/index.md [hunk 13]
```diff
@@ -0,0 +1,223 @@
...
+
+As you have seen, the circle() value creates a circular clipped area.
```
Comment (kaycebasques) [inline src/site/content/en/blog/css-clipping/index.md:105]:
```suggestion
As you have seen, the `circle()` value creates a circular clipped area.
```

File: src/site/content/en/blog/css-clipping/index.md [hunk 14]
```diff
@@ -0,0 +1,223 @@
...
+
+#### Watch out for flat edges!
+
+Be aware with all of these values that the shape will be clipped by the margin box on the element.
+If you create a circle on an image,
+and that shape would extend outside of the natural size of the image, you will get a flat edge.
+
+<figure class="w-figure">
+  <img class="w-screenshot" style="max-inline-size: 480px" src='./flat-edges.jpg' alt="A clipped circle with flat edges">
+  <figcaption>The image used earlier now has circle(50%) applied. As the image is not square, we hit the margin box at the top and bottom and the circle is clipped.</figcaption>
```
Comment (kaycebasques) [inline src/site/content/en/blog/css-clipping/index.md:128]:
```suggestion
  <figcaption>The image used earlier now has <code>circle(50%)</code> applied. As the image is not square, we hit the margin box at the top and bottom and the circle is clipped.</figcaption>
```

File: src/site/content/en/blog/css-clipping/index.md [hunk 15]
```diff
@@ -0,0 +1,223 @@
...
+</figure>
+
+### ellipse()
```
Comment (kaycebasques) [inline src/site/content/en/blog/css-clipping/index.md:131]:
```suggestion
### `ellipse()`
```

File: src/site/content/en/blog/css-clipping/index.md [hunk 16]
```diff
@@ -0,0 +1,223 @@
...
+<div class="glitch-embed-wrap" style="height: 420px; width: 100%;">
+  <iframe
+    src="https://glitch.com/embed/#!/embed/clip-path-ellipse?path=README.md&previewSize=100"
+    title="clip-path-ellipse on Glitch"
+    allow="encrypted-media"
+    style="height: 100%; width: 100%; border: 0;">
+  </iframe>
+</div>
+
+### polygon()
```
Comment (kaycebasques) [inline src/site/content/en/blog/css-clipping/index.md:146]:
```suggestion
### `polygon()`
```

File: src/site/content/en/blog/css-clipping/index.md [hunk 17]
```diff
@@ -0,0 +1,223 @@
...
+
+The `polygon()` value can help you to create fairly complex shapes,
```
Comment (kaycebasques) [inline src/site/content/en/blog/css-clipping/index.md:148]:
```suggestion
The `polygon()` value can help you create fairly complex shapes,
```

File: src/site/content/en/blog/css-clipping/index.md [hunk 18]
```diff
@@ -0,0 +1,223 @@
...
+<div class="glitch-embed-wrap" style="height: 420px; width: 100%;">
+  <iframe
+    src="https://glitch.com/embed/#!/embed/clip-path-polygon?path=README.md&previewSize=100"
+    title="clip-path-polygon on Glitch"
+    allow="encrypted-media"
+    style="height: 100%; width: 100%; border: 0;">
+  </iframe>
+</div>
+
+To help you create polygons and see what is possible check out Clippy,
```
Comment (kaycebasques) [inline src/site/content/en/blog/css-clipping/index.md:161]: Link to Clippy, please

File: src/site/content/en/blog/css-clipping/index.md [hunk 19]
```diff
@@ -0,0 +1,223 @@
...
+These values can be used alone, or alongside a basic shape to define the reference box used by the shape.
+For example, the following would clip the shape to the edge of the content.
+
+```css
+.box {
+  clip-path: content-box;
+}
+```
+
+In this example the circle would use the content-box as the reference box rather than the margin-box (which is the default).
```
Comment (kaycebasques) [inline src/site/content/en/blog/css-clipping/index.md:180]:
```suggestion
In this example the circle would use the `content-box` as the reference box rather than the `margin-box` (which is the default).
```

**Review by kaycebasques — CHANGES_REQUESTED** [review]
Looks great. Very polished. Just a bunch of minor requested changes.

File: src/site/content/en/blog/css-clipping/index.md (hunk 8 above)
Comment (rachelandrew) [reply src/site/content/en/blog/css-clipping/index.md:47]: the way you tend to use feature queries however is to do your old layout, then overwrite it in the query so it doesn't matter that IE11 neither supports the feature or the feature query as it ignores the whole lot.

File: src/site/content/en/blog/sign-in-form-best-practices/index.md (hunk 5 above)
Comment (rachelandrew) [reply src/site/content/en/blog/sign-in-form-best-practices/index.md:50]: I'm not sure what the change is here?

File: src/site/content/en/blog/css-clipping/index.md [hunk 20]
```diff
@@ -0,0 +1,224 @@
+---
+title: How to use CSS clipping
+description: |
+  Using clipping in CSS can help us move away from everything in our designs looking like a box.
+  By using various basic shapes, or an SVG, you can create a clip path.
+  Then cut away the parts of an element you don't want to show.
+authors:
+  - rachelandrew
+date: 2020-09-02
```
Comment (kaycebasques) [inline src/site/content/en/blog/css-clipping/index.md:9]:
```suggestion
date: 2020-09-14
```

File: src/site/content/en/blog/css-masking/index.md [hunk 21]
```diff
@@ -0,0 +1,196 @@
+---
+title: How to use CSS masking
+description: |
+  CSS masking gives you the option of using an image as a mask layer.
+  This means that use can use an image, an SVG, or a gradient as your mask,
+  to create interested effects without an image editor.
+authors:
+  - rachelandrew
+date: 2020-09-02
```
Comment (kaycebasques) [inline src/site/content/en/blog/css-masking/index.md:9]:
```suggestion
date: 2020-09-14
```

**Review by kaycebasques — APPROVED** [review]
